src/pages/home/report/EmojiPickerMenu.js

```diff
@@ -0,0 +1,111 @@
+import TextInputFocusable from '../../../components/TextInputFocusable';
+
+const propTypes = {
+    // Controls the visibility of this component.
+    isVisible: PropTypes.bool,
+    addEmojiToTextBox: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+    isVisible: false,
+};
+
+class EmojiPickerMenu extends Component {
+    constructor(props) {
+        super(props);
+
+        this.filterEmojis = _.debounce(this.filterEmojis.bind(this), 500, false);
+        this.state = {
+            filteredEmojis: emojis,
+            headerIndices: [0, 35, 61, 89, 100, 122, 149],
+        };
+    }
+
+    componentDidUpdate() {
+        if (this.props.isVisible) {
+            this.searchInput.focus();
+        }
+    }
+
+    filterEmojis(searchTerm) {
+        const normalizedSearchTerm = searchTerm.toLowerCase();
+        if (normalizedSearchTerm === '') {
+            this.setState({filteredEmojis: emojis});
+            return;
+        }
+        const newFilteredEmojiList = [];
+        emojis.forEach((emoji) => {
+            if (!emoji.header && emoji.code !== 'BLANK') {
+                emoji.keywords.forEach((keyword) => {
+                    if (keyword.includes(normalizedSearchTerm)) {
+                        newFilteredEmojiList.push(emoji);
+                    }
+                });
+            }
+        });
+
+        this.setState({filteredEmojis: newFilteredEmojiList});
+    }
+
+    renderItem({item}, addEmojiToTextBox) {
+        if (item.code === 'BLANK') {
+            return;
+        }
+
+        if (item.header) {
+            return (
+                <Text style={styles.emojiHeaderStyle}>
+                    {item.code}
+                </Text>
+            );
+        }
+
+        return (
+            <EmojiPickerMenuItem
+                onPress={addEmojiToTextBox}
+                emoji={item.code}
+            />
+        );
+    }
+
+    render() {
+        return (
+            this.props.isVisible && (
+                <View style={[styles.emojiPickerBox]}>
+                    <View style={[styles.pt4, styles.ph4]}>
+                        <TextInputFocusable
+                            textAlignVertical="top"
+                            placeholder="Search"
+                            placeholderTextColor={themeColors.textSupporting}
+                            onChangeText={this.filterEmojis}
+                            style={[styles.textInput]}
+                            defaultValue=""
+                            ref={el => this.searchInput = el}
+                        />
+                    </View>
+                    <FlatList
+                        data={this.state.filteredEmojis}
+                        renderItem={item => this.renderItem(item, this.props.addEmojiToTextBox)}
+                        keyExtractor={item => (`emoji_picker_${item.code}`)}
+                        numColumns={8}
+                        style={{height: 300, paddingHorizontal: 16, paddingBottom: 16}}
```

Comment:
We might want to move these styles out into styles.js?

Reply:
Also, see my comments in the main PR thread but I think we need to give this some kind of width too. I'm not sure the best approach that works well here but also is 100% wide on mobile too.